openmc/material.py

```diff
@@ -194,10 +194,6 @@ def distrib_otf_file(self):
     def id(self, material_id):
         global AUTO_MATERIAL_ID, MATERIAL_IDS
 
-        # If the Material already has an ID, remove it from global list
-        if hasattr(self, '_id') and self._id is not None:
-            MATERIAL_IDS.remove(self._id)
```

Comment:
Sure, the short story is that the global `MATERIAL_IDS` was from an early stage of the API when I wanted to prevent users from creating multiple `Material`, `Cell`, etc. objects with the same IDs. We relaxed this constraint for `Surfaces`, `Cells`, `Universes` and `Lattices` several months ago, but for some reason the constraint was kept around for `Materials` (perhaps there was a good reason for this that is slipping my mind now). Anyway, removing `MATERIAL_IDS` here is streamlines the `id` setter for `Material` with all of the other classes in the API.

This constraint on the `Material` IDs posed an issue for me recently when I was loading in materials from "summary.h5." The `Summary` API previously set the private `_id` attributes directly, and hence the IDs were not stored in `MATERIAL_IDS`. This caused line 199 above to error out when I tried to over-ride a deep copied `Material's` ID using the `id` setter. All of this led me to remove `MATERIAL_IDS` here and use the `id` setters in the `Summary` API.

Do you have reason to keep `MATERIAL_IDS` around @paulromano?